website/src/components/Edit/SequencesForm.tsx

```diff
@@ -126,6 +157,20 @@ export const SequencesForm: FC<SequenceFormProps> = ({ editableSequences, setEdi
                                     : undefined
                             }
                             showUndo={true}
+                            onDownload={
+                                field.initialValue.length > 0 && dataToEdit
+                                    ? () => {
+                                          const accessionVersion = `${dataToEdit.accession}.${dataToEdit.version}`;
+                                          generateAndDownloadFastaFile(
+                                              accessionVersion,
+                                              field.initialValue,
+                                              field.key,
```

Comment:
From the code above it looks like field.key always exists -> I think we can then simplify the `generateAndDownloadFastaFile` function.

Also as a side note - where is the key defined to be the segmentName? It might be nice to add that somewhere as a comment at least

Reply:
Typescript would generally tell us if we were doing a pointless comparison - I expect we are passing on something that could be undefined or null

Reply:
The key thing does seem confusing but it's not something I've introduced here and I'd prefer if we tackle it separately as I'm not sure exactly where it comes from and where we should document this